src/createSettingsFileTransformer.js

```diff
@@ -0,0 +1,162 @@
+var objectAssign = require('@adobe/reactor-object-assign');
+
+function isArrayReference(str) {
+  return isString(str) && str.indexOf('[') !== -1 && str.indexOf(']') !== -1;
+}
+function sanitizeArrayKey(pathStrSegment) {
+  return pathStrSegment.substr(
+    0,
+    // the name goes up to the start of the array bracket: 'someArrayName[]'
+    pathStrSegment.indexOf('[')
+  );
+}
+
+function isObject(value) {
```

Comment:
Last time we talked I saw that for the syntethic event we are using a lib called is-plain-object. Can't we use it here also and not duplicate the functionality?

Reply:
oh wonderful, I actually missed that!